wdl-format/src/token/pre.rs

```diff
@@ -0,0 +1,75 @@
+//! Tokens emitted during the formatting of particular elements.
+
+use wdl_ast::SyntaxKind;
+
+use crate::Token;
+use crate::TokenStream;
+
+/// A token that can be written by elements.
+///
+/// These are tokens that are intended to be written directly by elements to a
+/// [`TokenStream`](super::TokenStream) consisting of [`PreToken`]s. Note that
+/// this will transformed into a [`TokenStream`](super::TokenStream) of
+/// [`PostToken`](super::PostToken)s by a
+/// [`Postprocessor`](super::Postprocessor) (authors of elements are never
+/// expected to write [`PostToken`](super::PostToken)s directly).
+#[derive(Debug, Eq, PartialEq)]
+pub enum PreToken {
+    /// A section spacer.
+    SectionSpacer,
+
+    /// Includes text literally in the output.
+    Literal(String, SyntaxKind),
+}
+
+impl PreToken {
+    /// Gets the [`SyntaxKind`] of the token if the token is a
+    /// [`PreToken::Literal`].
+    pub fn kind(&self) -> Option<&SyntaxKind> {
+        match self {
+            PreToken::Literal(_, kind) => Some(kind),
+            _ => None,
+        }
+    }
+}
+
+/// The line length to use when displaying pretokens.
+const DISPLAY_LINE_LENGTH: usize = 88;
```

Comment:
Why 88? Me and @adthrasher settled on 90 a very long time ago, though that was somewhat arbitrary. From my memory of the decision, many code formatters have settled that the "ideal" file width is somewhere around 85-90 characters wide, and me and Andrew opted for 90 as a round and easily remembered number.

Reply:
We can change this—I didn't give a lot of thought to it, it's just used for debugging.